lib/publiccloud/ssh_interactive.pm

```diff
@@ -52,10 +52,15 @@ sub ssh_interactive_tunnel {
 
     # Pipe the output of the device fifo to the local serial terminal
 # Note: We run this in a loop so that the ssh tunnel gets automatically re-established after device reboots and such. The sleep helps to avoid unnecessary CPU hogging in case of connection issues
-    enter_cmd("while true; do ssh sut -yt -R '$upload_port:$upload_host:$upload_port' 'rm -f /dev/sshserial && mkfifo -m a=rwx /dev/sshserial && tail -fn +1 /dev/sshserial' 2>&1 >/dev/$serialdev; sleep 5; done");
+    my $cmd = "while true; do ssh sut -yt -R '$upload_port:$upload_host:$upload_port' 'rm -f /dev/sshserial && mkfifo -m a=rwx /dev/sshserial && tail -fn +1 /dev/sshserial' 2>&1 >/dev/$serialdev; sleep 5; done";
+    enter_cmd($cmd);
     # give the ssh connection some time to settle
     sleep 10;
 
+    # Check if the connection is established
+    my $check = script_run('test -e /dev/sshserial');
```

Comment:
but from what I understand this won't help to address https://progress.opensuse.org/issues/182357 . job failed in https://progress.opensuse.org/issues/182357 just stuck in `enter_cmd` so in such a case you won't be able to reach this line

Reply:
after thinking about this second time , I think this check make sense :thinking:  @pdostal  WDYT ?

Reply:
https://progress.opensuse.org/issues/182948